lib/project_types/script/layers/infrastructure/script_project_repository.rb

```diff
@@ -77,22 +76,21 @@ def create_env(api_key:, secret:, uuid:)
             script_name: script_name,
             extension_point_type: extension_point_type,
             language: language,
-            script_json: ScriptJsonRepository.new(ctx: ctx).get,
+            script_config: script_config!,
           )
         end
 
-        def update_or_create_script_json(title:)
-          script_json = ScriptJsonRepository
-            .new(ctx: ctx)
-            .update_or_create(title: title)
+        def update_or_create_script_config(title:)
+          script_config = script_json_repository.update(title: title) ||
+            script_config_yml_repository.update_or_create(title: title)
```

Comment:
Related to my comment below, I don't think we need to specifically call the JSON or YAML implementations specifically here. I think we can use an instance of `ScriptConfigRepository`, which is backed by one of the concrete classes. Since all valid script projects should have a `script.json` or `script.config.yml`, I think we could expose an `update` method and use that.